src/states/elder/mod.rs

```diff
@@ -2084,6 +2098,12 @@ impl Approved for Elder {
         Ok(())
     }
 
+    fn handle_ack_message_event(&mut self, sec_info: SectionInfo) -> Result<(), RoutingError> {
+        self.chain
+            .update_their_knowledge(*sec_info.prefix(), *sec_info.version());
```

Comment:
Funny, I did not expect this function to exist.
I think we need to update what that function to do the right thing.
Also not clear if https://github.com/maidsafe/routing/issues/1673 was really done, and maybe you need it here.

Also  update_their_knowledge is called when NeighbourConfirm is called. (Wondering whether this AckMessage is not kind of trying to replace NeighbourConfirm. Anyway, I think it should no longer be called in NieghbourConfirm.